core/pkg/sync/http/http_sync.go

```diff
@@ -9,25 +9,79 @@ import (
 	"io"
 	"net/http"
 	parseUrl "net/url"
+	"os"
 	"path/filepath"
+	"time"
 
 	"github.com/open-feature/flagd/core/pkg/logger"
 	"github.com/open-feature/flagd/core/pkg/sync"
 	"github.com/open-feature/flagd/core/pkg/utils"
+	"github.com/robfig/cron"
+	"go.uber.org/zap"
 	"golang.org/x/crypto/sha3" //nolint:gosec
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
 )
 
 type Sync struct {
-	URI         string
-	Client      Client
-	Cron        Cron
-	LastBodySHA string
-	Logger      *logger.Logger
-	BearerToken string
-	AuthHeader  string
-	Interval    uint32
+	uri         string
+	client      Client
+	cron        Cron
+	lastBodySHA string
+	logger      *logger.Logger
+	bearerToken string
+	authHeader  string
+	interval    uint32
 	ready       bool
 	eTag        string
+
+	oauthCredential *oauthCredentialHandler
+}
+
+type oauthCredentialHandler struct {
+	clientId           string
+	clientSecret       string
+	tokenUrl           string
+	folderSource       string
+	reloadDelaySeconds int
+
+	lastUpdate time.Time
+}
+
+func (och *oauthCredentialHandler) loadOAuthConfiguration() (*clientcredentials.Config, error) {
+	if och.folderSource == "" {
+		oauth := &clientcredentials.Config{
+			ClientID:     och.clientId,
+			ClientSecret: och.clientSecret,
+			TokenURL:     och.tokenUrl,
+			AuthStyle:    oauth2.AuthStyleInParams,
+		}
+		return oauth, nil
+	}
+	// we load from files
+	id, err := os.ReadFile(och.folderSource + "/" + och.clientId)
```

Comment:
to avoid too many params, I am reusing the `clientId` as filename

Reply:
could we actually use just the following
```suggestion
	id, err := os.ReadFile(och.folderSource + "/clientId")
```
we can decide via config anyways where the secrets are stored, and this way we do not multipurpose the clientId field (sidenote same for ClientSecret).

// pair-review with @alexandraoberaigner @guidobrei